Core/include/Acts/TrackFitting/GlobalChiSquareFitter.hpp

```diff
@@ -524,12 +589,14 @@ class Gx2Fitter {
     using PropagatorOptions = Acts::PropagatorOptions<Actors, Aborters>;
 
     const size_t reducedMatrixSize = 4;
-    Acts::CurvilinearTrackParameters params = sParameters;
+    start_parameters_t params = sParameters;
     BoundVector deltaParams = BoundVector::Zero();
     double chi2sum = 0;
     BoundMatrix aMatrix = BoundMatrix::Zero();
     BoundVector bVector = BoundVector::Zero();
 
+    GX2FResult gx2fResult;
```

Comment:
Default constructed here and then assigned (copied?) later? Maybe this should be an optional and receive the target? Not sure if that's cleaner.

Reply:
We need to pre-construct it because we write to it during each update. It stores the propagation result, which will be used in future iterations. We also need it to have access to the track outside the update-loop. (Needed for the final trajectory)